certmanager/ca.go

```diff
@@ -50,22 +63,36 @@ func NewSelfSignedIstioCA() *IstioCA {
 		IsSelfSigned: true,
 	}
 	pemCert, pemKey := GenCert(options)
-	cert, key := parsePemEncodedCertificateAndKey(pemCert, pemKey)
 
-	return NewIstioCA(cert, key)
+	opts := &IstioCAOptions{
+		CertChainBytes:   pemCert,
```

Comment:
In self-signed root certificate case, you don't need to include the root cert here.

Basically the only certificates needed to be included are the one before root certificate, since the root certificate is already distributed to the other end.

https://tools.ietf.org/html/rfc5246#section-7.4.2

It is just an optimization, it is ok to include, but not necessary.